smac/facade/hydra_facade.py

```diff
@@ -222,37 +177,33 @@ def optimize(self) -> typing.List[Configuration]:
             self.logger.info("="*120)
             self.logger.info("Hydra Iteration: %d", (i + 1))
 
-            # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Multiprocessing part start ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ #
-            q = multiprocessing.Queue()
-            procs = []
-            for p in range(self.n_optimizers):
-                proc = multiprocessing.Process(target=optimize,
-                                               args=(
-                                                   q,                   # Output queue
-                                                   self.scenario,       # Scenario object
-                                                   self._tae,           # type of tae to run target with
-                                                   p,                   # process_id (used in output folder name)
-                                                   self.output_dir,     # directory to create outputs in
-                                                   self.cost_per_inst   # portfolio cost per instance
-                                               ),
-                                               kwargs=self.kwargs)
-                proc.start()
-                procs.append(proc)
-            for proc in procs:
-                proc.join()
-            incs = np.empty((self.n_optimizers, ), dtype=Configuration)
-            self.logger.info('*'*120)
-            self.logger.info('Incumbents this round:')
-            idx = 0
-            while not q.empty():
-                conf = q.get_nowait()
-                self.logger.info(conf)
-                incs[idx] = conf
-                idx += 1
-            q.close()
-            # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Multiprocessing part end ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ #
-
-            cur_portfolio_cost = self._update_portfolio(incs)
+            self.optimizer = PSMAC(
+                scenario=self.scenario,
+                run_id=self.run_id,
+                rng=self.rng,
+                tae=self._tae if i == 0 else partial(ExecuteTARunHydra, cost_oracle=self.cost_per_inst),
+                shared_model=False,
+                validate=True if self.val_set else False,
+                n_optimizers=self.n_optimizers,
+                val_set=self.val_set,
+                n_incs=self.n_optimizers,  # return all configurations (unvalidated)
+                **self.kwargs
+            )
+            self.optimizer.output_dir = self.output_dir
+            incs = self.optimizer.optimize()
+            cost_per_conf_v, val_ids, cost_per_conf_e, est_ids = self.optimizer.get_best_incumbents_ids(incs)
+            if self.val_set:
+                to_keep_ids = val_ids[:self.incs_per_round]
+            else:
+                to_keep_ids = est_ids[:self.incs_per_round]
+            config_cost_per_inst = {}
+            incs = incs[to_keep_ids]
+            self.logger.info('Kept incumbents')
```

Comment:
I do not really understand what this message means. Do you mean "Kept X incumbents" or "Kept incumbents this round" ?
Could you make this message more meaningful or remove it?